src/components/has-searched.mixin.vue

```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
```

Comment:
You can write this `mixin` in a `.ts` file instead of a `.vue` one, as it is just a `class` and does not have neither a template nor styles, and thus remove the `<script lang="ts" />` tag. This is how we have most of the `mixins` defined in `x-components`